src/main/java/in/gov/cdc/cdcdemostage/validators/dob/DobObjectValidator.java

```diff
@@ -0,0 +1,142 @@
+     * the date object.
+     * TODO: to be decorated with the loggers
+     * @param extractedPacket
+     * @param event
+     * @return
+     */
+    @Override
+    public Optional<List<ValidationError>> validate(ExtractedPacket extractedPacket, Event event) {
+
+        List<ValidationError> errors = new ArrayList<>();
+
+
+        syntaxValidation(extractedPacket, event).ifPresent(errors::addAll);
+
+        if (errors.isEmpty()) {
+            semanticValidation(extractedPacket, event).ifPresent(errors::addAll);
+            policyValidation(extractedPacket, event).ifPresent(errors::addAll);
+        }
+
+        if (errors.isEmpty()) {
+            // All validations passed
+            System.out.println("DOB validation completed!!");
+            return Optional.empty();
+        } else {
+            // TODO: Over here reduce the errors to an existing SEDA error code related DoB and set the rejected bit indicator
+            return Optional.of(errors);
+        }
+    }
+
+    /**
+     * This method performs the syntax validation on the Dob
+     * @param packet
+     * @param event
+     * @return
+     */
+    public Optional<List<ValidationError>> syntaxValidation(ExtractedPacket packet, Event event) {
+        return Optional.of(getSupportedValidators(ISyntaxValidator.class,
+                event.getRequestBitmap()).stream()
+                .flatMap(v -> v.validate(packet).stream())
+                .collect(Collectors.toList()));
+
+    }
+
+    /**
+     * This class performs semantic validation on the DoB
+     *
+     * @param packet
+     * @return
+     */
+    public Optional<List<ValidationError>> semanticValidation(ExtractedPacket packet, Event event) {
+
+        return Optional.of(getSupportedValidators(ISemanticValidator.class,
+                event.getRequestBitmap()).stream()
+                .flatMap(v -> v.validate(packet).stream())
+                .collect(Collectors.toList()));
+
+    }
+
+    /**
+     * This class performs policy validation on the Dob
+     *
+     * @param packet
+     * @return
+     */
+    public Optional<List<ValidationError>> policyValidation(ExtractedPacket packet, Event event){
+
+        return Optional.of(getSupportedValidators(IPolicyValidator.class,
+                event.getRequestBitmap()).stream()
+                .flatMap(v -> v.validate(packet).stream())
+                .collect(Collectors.toList()));
+    }
+
+
+    // Given an event message, return a bitset of the current change
+//    private BitSet filter(EventMessage event) {
+//        BitSet bs = event.getUidV2DataArray()[0].getUidOriginTracker().getUpdateBits();
+//        bs.xor(event.getUidV2DataArray()[0].getUidOriginTracker().getRejectBits());
+//        System.out.println("bitset : " + bs);
+//        return bs;
+//    }
+
+    // Given a bitset and validators get all the supported validators
+    private <T> List<? extends IValidator> getSupportedValidators(Class<T> validatorType, BitSet bitSet) {
```

Comment:
This should happen in the constructor

Reply:
Not understood.

Reply:
What I meant is, this function can be called in constructor after we get the list of validators. Same as this comment
https://github.com/Vajra456/cdc-demo-stage/pull/1#discussion_r1423054995